server/src/api/taskratings/taskratings.controller.ts

```diff
@@ -186,13 +186,23 @@ export const patchTaskratings: RouteHandlerFnc = async (ctx) => {
   if (foundRatings.some(({ err }) => err === 403))
     return void (ctx.status = 403);
 
-  const updated = await Taskrating.transaction((trx) =>
-    Promise.all(
+  const result = await Taskrating.transaction(async (trx) => {
+    const toDelete = foundRatings.filter(({ value }) => !value);
+    const deleted = await Promise.all(
+      toDelete.flatMap(({ rating }) =>
+        rating ? Taskrating.query(trx).findById(rating.id).delete() : [],
+      ),
+    );
```

Comment:
```suggestion
   const toDelete = foundRatings.flatMap(({ value, rating }) =>
     !value && rating ? [rating.id] : [],
   );
   const deleted = await Taskrating.query(trx).findByIds(toDelete).delete();
```

This would be better to do the other way around: a single query with multiple ids.